### PR TITLE
Optimize Levenshtein implementation

### PR DESCRIPTION
**Pull request by megawac**
This implementation is directly from https://github.com/hiddentao/fast-levenshtein

Before this change:
```
[21:17:45] levenshtein x 25,172 ops/sec ±0.70% (98 runs sampled)
```

After:
```
[21:09:18] levenshtein x 141,238 ops/sec ±0.95% (97 runs sampled)
```

Relevant jsperf: http://jsperf.com/levenshtein-distance/22 (I experimented with some of the others but this one had by far the best performance in node)

### CODE REVIEW

File: levenshtein.js [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
-      current[j] = value;
+  // calculate current row distance from previous row
+  for (i = 0; i < str1.length; ++i) {
+    var nextCol = i + 1;
+
+    for (var j = 0; j < str2.length; ++j) {
+      var curCol = nextCol;
+
+      // substution
+      nextCol = prevRow[j] + ( (str1.charAt(i) === str2.charAt(j)) ? 0 : 1 );
```
Comment (megawac) [inline levenshtein.js]: The ternary is optional here; coercing to a boolean will work as well

**Comment by stoeffel** [comment]
@megawac sure I will have a look at this tomorrow and publish a new release.

**Comment by stoeffel** [comment]
how about we link to hiddentao since this is directly from his implementation?

**Comment by stoeffel** [comment]
otherwise LGTM. I will check it once again this evening and then hopefully have time to pump the version.

**Comment by stoeffel** [comment]
@megawac I published 3.1.0

**Comment by megawac** [comment]
Sweetness thanks!

**Comment by stoeffel** [comment]
Just had to publish again since I accidentally pushed the `coverage` folder to npm.

**Pull request merged**